core/src/main/scala/org/apache/spark/rdd/RDDOperationScope.scala

```diff
@@ -95,17 +95,17 @@ private[spark] object RDDOperationScope extends Logging {
   private[spark] def withScope[T](
       sc: SparkContext,
       allowNesting: Boolean = false)(body: => T): T = {
-    val stackTrace = Thread.currentThread.getStackTrace().tail // ignore "Thread#getStackTrace"
-    val ourMethodName = stackTrace(1).getMethodName // i.e. withScope
-    // Climb upwards to find the first method that's called something different
-    val callerMethodName = stackTrace
-      .find(_.getMethodName != ourMethodName)
+
+    val callerMethodName = Thread.currentThread.getStackTrace()
+      .dropWhile(! _.getMethodName().equals("withScope"))
```

Comment:
I get it, I'm reading it correctly at last now. Minor: you have two different ways of expressing the same check in these two lines; can they both be `_.getMethodName != "withScope"`?